### ISSUE TITLE
Upgrade tip for previous HyprV users

### ISSUE THREAD

**Issue by JesseKoldewijn** [SolDoesTech/HyprV4#1]
Hey @SolDoesTech first of all, thanks for the improved script flow. 

Wanted to give the following tip to people that might upgrade to v4 (while using a nvidia gpu) from one of the previous revisions of HyprV. Make sure to remove the "hyprland" package from your system beforehand. Having this package installed beforehand kills the script on the step where it will try to install "hyprland-nvidia".

Alongside this tip, I'll be forking this repo where I will replace yay with paru, fish (shell) and kitty with alacritty again. I'll also throw in a config for both alacritty, starship and fish (aliases for curtain commands).

Cheers!🤙

**Comment by JesseKoldewijn** [comment]
> Hello and thank you for the feedback. I just added a bit of logic to fix this so much appreciated! also I like what you are doing with your project. I am starting to think that this "simple" script may need to have a single home and not multiple repos and with a more define upgrade process and options for the users. Once I put that together ill let you know if you would be interested in adding some of your enhancements in. cheers mate!

Hey man, I think one of the improvements could be with the move to paru.

One of the nice parts about paru is when using it in a script it doesnt die when you use noconfirm (which is sort of equal to skipreview in paru) because it would resolve the clashing package and ask you if you like to replace it. 

By having this prompt in the script it would make the upgrade flow a little easier to maintain I think.🤙

**Comment by thanhquang1988** [comment]
Thank you very much. I have one question, I installed v3 already, how to fix the issue with capture screen without reinstall v4?

**Comment by JesseKoldewijn** [comment]
> Thank you very much. I have one question, I installed v3 already, how to fix the issue with capture screen without reinstall v4?

What do you mean by capture screen? If you're referring to screen sharing on applications like Discord/Teams etc. this wont really work since this is still kind of broken on wayland as far as I've experienced. This is why I do have kde installed besides my hyprland environment to have some form of a desktop environment that runs on X11 instead of wayland.

Keep in mind, wayland is still way less mature than X11 but has mainly less screen tearing, smoother animations etc.
This would definitely change in the future but for now screen sharing won't really work all to great.

**Comment by thanhquang1988** [comment]
> > Thank you very much. I have one question, I installed v3 already, how to fix the issue with capture screen without reinstall v4?
> 
> 
> 
> What do you mean by capture screen? If you're referring to screen sharing on applications like Discord/Teams etc. this wont really work since this is still kind of broken on wayland as far as I've experienced. This is why I do have kde installed besides my hyprland environment to have some form of a desktop environment that runs on X11 instead of wayland.
> 
> 
> 
> Keep in mind, wayland is still way less mature than X11 but has mainly less screen tearing, smoother animations etc.
> 
> This would definitely change in the future but for now screen sharing won't really work all to great.

Sorry I mean black screenshot issue

**Comment by JesseKoldewijn** [comment]
> > > Thank you very much. I have one question, I installed v3 already, how to fix the issue with capture screen without reinstall v4?
> > 
> > 
> > 
> > What do you mean by capture screen? If you're referring to screen sharing on applications like Discord/Teams etc. this wont really work since this is still kind of broken on wayland as far as I've experienced. This is why I do have kde installed besides my hyprland environment to have some form of a desktop environment that runs on X11 instead of wayland.
> > 
> > 
> > 
> > Keep in mind, wayland is still way less mature than X11 but has mainly less screen tearing, smoother animations etc.
> > 
> > This would definitely change in the future but for now screen sharing won't really work all to great.
> 
> Sorry I mean black screenshot issue

Thats mainly a wayland issue. Just like with screen recording + sharing. You can fix it but its quite a hastle which will not ensure it keeps working.

**Comment by thanhquang1988** [comment]
> > > > Thank you very much. I have one question, I installed v3 already, how to fix the issue with capture screen without reinstall v4?
> > > 
> > > 
> > > What do you mean by capture screen? If you're referring to screen sharing on applications like Discord/Teams etc. this wont really work since this is still kind of broken on wayland as far as I've experienced. This is why I do have kde installed besides my hyprland environment to have some form of a desktop environment that runs on X11 instead of wayland.
> > > Keep in mind, wayland is still way less mature than X11 but has mainly less screen tearing, smoother animations etc.
> > > This would definitely change in the future but for now screen sharing won't really work all to great.
> > 
> > 
> > Sorry I mean black screenshot issue
> 
> Thats mainly a wayland issue. Just like with screen recording + sharing. You can fix it but its quite a hastle which will not ensure it keeps working.

Yes, I have just installed the hyprland-nvidia package, and it is now working.
Thank you again.

**Comment by SolDoesTech** [comment]
the hyprland-nvidia would resolve the issue but you also want to take a look at the hyprland.conf file and the associated env_var.conf and env_var_nvidia.conf to make sure things are setup correctly (all located in ~/.config/hypr/. you may also need to update your ~/.start-hypr-nvidia script but that part is optional.
